kinesis/kinesis.go

```diff
@@ -196,8 +196,9 @@ func (outputPlugin *OutputPlugin) AddRecord(record map[interface{}]interface{},
 	}
 
 	newDataSize := len(data)
```

Comment:
Add the `len(partitionKey)` here so that this var is the actual "new data size"

Reply:
Also, then you should error and discard the record if `newDataSize` > 1 MiB. That's the max size for a record. As far as I can tell we do not currently check for that in this plugin..

Reply:
> Add the `len(partitionKey)` here so that this var is the actual "new data size"

I intentionally kept them separated so that the next developer can easily understand how the actual dataLength is calculated. 
dataLength = newDataSize + partitionKeySize

Reply:
Ok

Reply:
I think you could call it `newRecordSize` though and they could see that the value is `len(data) + len(partitionKey)` which would accomplish the same thing with less code...

Reply:
> Also, you should error and discard the record if `newDataSize > 1 MiB`. That's the max size for a record. As far as I can tell we do not currently check for that in this plugin..

Good point. +1
I added another commit.